src/main/kotlin/nl/avisi/kotlinwebtest/jdbc/Database.kt

```diff
@@ -0,0 +1,25 @@
+/**
+ * For licensing, see LICENSE.txt
+ * @author Martijn Heuvelink
+ */
+package nl.avisi.kotlinwebtest.jdbc
+
+import nl.avisi.kotlinwebtest.StepRequest
+import nl.avisi.kotlinwebtest.StepResponse
+
+abstract class DatabaseRequest : StepRequest {
+    var driver: String? = null
+    var url: String? = null
+    var query: String? = null
+
+    abstract infix fun url(url: String)
+    abstract infix fun driver(driver: String)
+
+    infix fun query(query: String) {
+        this.query = query
+    }
+}
+
+interface DatabaseStepResponse : StepResponse {
+    val resultSet: List<HashMap<String, String>>?
```

Comment:
geen optional proberen te gebruiken hiervoor, want dan heb je 3 state, no list, empty list of filled list